InteractiveHtmlBom/ecad/fusion_eagle.py

```diff
@@ -0,0 +1,618 @@
+                    pad['shape'] = 'rect'
+                else:
+                    pad['shape'] = 'roundrect'
+                    pad['radius'] = (float(el.attrib['roundness']) / 100) \
+                                    * float(el.attrib['dy']) / 2
+                pads.append(pad)
+        return pads
+
+    def _add_silkscreen(self, el, x, y, angle, mirrored, silkscreen):
+        if el.attrib['layer'] in [self.TOP_PLACE_LAYER, self.BOT_PLACE_LAYER]:
+            dwg = None
+
+            if el.tag in ['wire', 'rectangle']:
+                x1 = (x +
+                      float(el.attrib['x1']) * math.cos(math.radians(angle)) -
+                      float(el.attrib['y1']) * math.sin(math.radians(angle)))
+                x2 = (x +
+                      float(el.attrib['x2']) * math.cos(math.radians(angle)) -
+                      float(el.attrib['y2']) * math.sin(math.radians(angle)))
+                y1 = -(y +
+                       float(el.attrib['x1']) * math.sin(math.radians(angle)) +
+                       float(el.attrib['y1']) * math.cos(math.radians(angle)))
+                y2 = -(y +
+                       float(el.attrib['x2']) * math.sin(math.radians(angle)) +
+                       float(el.attrib['y2']) * math.cos(math.radians(angle)))
+
+                if el.tag == 'wire':
+                    dwg = {'type': 'segment',
+                           'start': [x1, y1],
+                           'end': [x2, y2],
+                           'width': float(el.attrib['width'])}
+                else:
+                    dwg = {'type': 'polygon',
+                           'filled': 1,
+                           'pos': [0, 0],
+                           'polygons': [[
+                               [x1, y1],
+                               [x2, y1],
+                               [x2, y2],
+                               [x1, y2]
+                           ]]
+                           }
+
+            elif el.tag == 'circle':
+                xc = (x +
+                      float(el.attrib['x']) * math.cos(math.radians(angle)) -
+                      float(el.attrib['y']) * math.sin(math.radians(angle)))
+                yc = -(y +
+                       float(el.attrib['x']) * math.sin(math.radians(angle)) +
+                       float(el.attrib['y']) * math.cos(math.radians(angle)))
+                dwg = {'type': 'circle',
+                       'start': [xc, yc],
+                       'radius': float(el.attrib['radius']),
+                       'width': float(el.attrib['width'])}
+
+            if dwg is not None:
+                top = el.attrib['layer'] == self.TOP_PLACE_LAYER
+                bot = el.attrib['layer'] == self.BOT_PLACE_LAYER
+
+                if (mirrored and bot) or (not mirrored and top):
+                    silkscreen['F'].append(dwg)
+                elif (mirrored and top) or (not mirrored and bot):
+                    silkscreen['B'].append(dwg)
```

Comment:
This looks fishy. If drawing is on bot but is mirrored then you add it to top silkscreen?
Is that how eagle plots gerbers too?

I guess this means drawing is on bot originally but since whole footprint is flipped (mirrored) it ends up on top? If so then the `mirrored` variable here and and in related functions should be `flipped`, unless this is the native terminology used in eagle.

Reply:
'mirrored' is the native terminology in Eagle for 'flipped to the other side of the board' (all relative to what side of the board any given element was on in the footprint in the component library) - example of a resistor 'mirrored' to the bottom of the board (blue=bottom):
<img width="251" alt="image" src="https://user-images.githubusercontent.com/24237058/197659303-2e977a83-0fe9-401d-88c4-2dd58df7a28b.png">

Conventionally part footprints in libraries are designed on top by default, and most often parts only have silkscreen on the same side as the part itself, but occasionally I do see a part that has silkscreen on both sides (e.g. a connector,  where it's handy to have a pin 1 indication on the back side for probing purposes)

In any case, I agree that the nomenclature is a little less than intuitive so I'll at least add a clarifying comment for posterity.